arbitrum-client/src/client/contract_interaction.rs

```diff
@@ -0,0 +1,156 @@
+//! Defines `ArbitrumClient` helpers that allow for interacting with the
+//! darkpool contract
+
+use circuit_types::{merkle::MerkleRoot, wallet::Nullifier};
+use constants::{Scalar, ScalarField, SystemProof};
+use renegade_contracts_common::{
+    serde_def_types::SerdeScalarField,
+    types::{ValidWalletCreateStatement, ValidWalletUpdateStatement, ValidMatchSettleStatement},
+};
+
+use crate::{
+    conversion::ToContractType,
+    errors::ArbitrumClientError,
+    helpers::{deserialize_retdata, serialize_calldata}, types::MatchPayload,
+};
+
+use super::ArbitrumClient;
+
+// TODO: Replace `renegade_contracts_common::types::*` with relayer statement
+// types once they're adapted to Plonk
+
+impl ArbitrumClient {
+    // -----------
+    // | GETTERS |
+    // -----------
+
+    /// Get the current Merkle root in the contract
+    pub async fn get_merkle_root(&self) -> Result<ScalarField, ArbitrumClientError> {
+        let merkle_root_bytes = self
+            .darkpool_contract
+            .get_root()
+            .call()
+            .await
+            .map_err(|e| ArbitrumClientError::ContractInteraction(e.to_string()))?;
+
+        Ok(deserialize_retdata::<SerdeScalarField>(&merkle_root_bytes)?.0)
+    }
+
+    /// Check whether the given Merkle root is a valid historical root
+    pub async fn check_merkle_root_valid(
+        &self,
+        root: MerkleRoot,
+    ) -> Result<bool, ArbitrumClientError> {
+        self.darkpool_contract
+            .root_in_history(serialize_calldata(&SerdeScalarField(root.inner()))?)
+            .call()
+            .await
+            .map_err(|e| ArbitrumClientError::ContractInteraction(e.to_string()))
```

Comment:
nit: but i think this is something that should be done more generally; these one liners with lots of nested expressions are hard to read and reason about, I suggest:
```rust
let calldata = serialize_calldata(...);
self.darkpool_contract.root_in_history(calldata)...
```
Same with below; it is in general hard to read to put everything in a chain of method calls.